cupy/core/raw.pyx

```diff
@@ -105,6 +105,36 @@ cdef class RawKernel:
             self._kernel_cache[dev] = ker
         return ker
 
+    # It is not possible to implement __reduce__ for a cdef class. The
+    # two-tuple return cannot handle the keyword-only arguments, and
+    # the three-tuple return (for updating the object's internal state)
+    # does not work either, because cdef classes by default does not have
+    # __dict__. Therefore, the only way to handle keyword-only arguments
+    # for picking a cdef class is to define the following two special
+    # functions, which is in fact preferred over __reduce__.
+
+    def __getstate__(self):
+        cdef dict args = {'code': self.code,
+                          'name': self.name,
+                          'options': self.options,
+                          'backend': self.backend,
+                          'translate_cucomplex': self.translate_cucomplex,
+                          'file_path': self.file_path,
```

Comment:
Then the file must be put in a globally visible place in the first place. This attribute is mainly for loading cubin/ptx from disk, so it's in a node-local scratch the user is doomed.

Reply:
This file path is where the compiled kernel is cached on disk right? Or is it something else?

Reply:
It's the cubin/ptx path that a user specifies would specify if they do `mod = cp.RawModule(path=file_path, ...); ker = mod.get_function(...)`.

Reply:
Oh, I see the confusion. CuPy's compiled kernel cache on disk is managed by `CUPY_CACHE_DIR`, not here.

Reply:
Ah ok. So this only really matters if a library is shipping their own PTX code? NVM then that makes sense :)

Just to make sure I understand correctly this would be `None` in Jens' case?

Reply:
Yes and yes! 👍

Reply:
Cool thanks for clarifying. Sorry for the confusion 😅

Reply:
No worries John. It's always good to have more eyes going over a change.